apps/hannk/delegate/hannk_delegate.cpp

```diff
@@ -1303,11 +1303,9 @@ class NodeSupport {
         if (!IsVersionOK(1, 2)) {
             return false;
         }
-
-        if (!InputHasType(1, I32)) {
+        if (!InputsHaveCorrectTypes({ANY, I32})) {
```

Comment:
The current helper funcs for detecting "is supported" feel poorly thought out to me (e.g. it was easy for you to make this mistake, which is not good), but not a lot of thought was put into it at the start. At this point though I wonder is switching to a table-driven approach would be worthwhile.

Reply:
I agree, but we shouldn't change it in this PR.

Reply:
Agreed, wasn't suggesting doing it here, just in general